experiments/AMIP/modular/components/ocean/eisenman_seaice.jl

```diff
@@ -0,0 +1,293 @@
+    z0b::FT = 1e-5                  # roughness length for buoyancy [m]
+    α::FT = 0.06                    # albedo
+end
+
+"""
+    state_init(p::EisenmanIceParameters, space)
+
+Initialize the state vectors for the Eisenman-Zhang sea ice model.
+"""
+function state_init(p::EisenmanIceParameters, space::Spaces.AbstractSpace)
+    Y = Fields.FieldVector(
+        T_sfc = ones(space) .* p.T_freeze,
+        h_ice = zeros(space),
+        T_ml = ones(space) .* 277,
+        q_sfc = ClimaCore.Fields.zeros(space),
+    )
+    Ya = Fields.FieldVector(
+        F_turb = ClimaCore.Fields.zeros(space),
+        ∂F_turb_energy∂T_sfc = ClimaCore.Fields.zeros(space),
+        F_rad = ClimaCore.Fields.zeros(space),
+        F_base = ClimaCore.Fields.zeros(space),
+        ocean_qflux = ClimaCore.Fields.zeros(space),
+        ρ_sfc = ClimaCore.Fields.zeros(space),
+    )
+    return Y, Ya
+end
+
+"""
+    get_∂F_rad_energy∂T_sfc(T_sfc, p)
+
+Calculate the derivative of the radiative flux with respect to the surface temperature.
+"""
+function get_∂F_rad_energy∂T_sfc(T_sfc, p)
+    FT = eltype(T_sfc)
+    @. FT(4) * (FT(1) - p.α) * p.σ * T_sfc^3
+end
+
+"""
+    solve_eisenman_model!(Y, Ya, p, Δt::FT)
+
+Solve the Eisenman-Zhang sea ice model for one timestep.
+"""
+function solve_eisenman_model!(Y, Ya, p, thermo_params, Δt)
+
+    # model parameter sets
+    (; p_i, p_o) = p
+
+    # prognostic
+    (; T_sfc, h_ice, T_ml, q_sfc) = Y
+
+    # auxiliary
+    (; F_turb, ∂F_turb_energy∂T_sfc, F_rad, ocean_qflux, F_base) = Ya
+
+    # local
+    F_atm = @. F_turb + F_rad
+    ∂F_atmo∂T_sfc = get_∂F_rad_energy∂T_sfc.(T_sfc, Ref(p_i)) .+ ∂F_turb_energy∂T_sfc
+
+    # ice thickness and mixed layer temperature changes due to atmosphereic and ocean fluxes
+    if parent(h_ice)[1] > 0 # ice-covered
+        @. F_base = p_i.C0_base * (T_ml - p_i.T_base)
+        ΔT_ml = @. -(F_base - ocean_qflux) * Δt / (p_o.h * p_o.ρ * p_o.c)
+        Δh_ice = @. (F_atm - F_base - ocean_qflux) * Δt / p_i.L_ice
+    else # ice-free
+        ΔT_ml = @. -(F_atm - ocean_qflux) * Δt / (p_o.h * p_o.ρ * p_o.c)
+        Δh_ice = 0
+    end
+
+    # T_ml is not allowed to be below freezing (i.e. ice-growth)
+    if (parent(T_ml .+ ΔT_ml)[1] < p_i.T_freeze)
+        Δh_ice = @. Δh_ice - (T_ml + ΔT_ml - p_i.T_freeze) * (p_o.h * p_o.ρ * p_o.c) / p_i.L_ice
+        ΔT_ml = @. p_i.T_freeze - T_ml
+    end
+
+    # adjust if transition to ice-free
+    if ((parent(h_ice)[1] > 0) & (parent(h_ice .+ Δh_ice)[1] <= 0))
+        ΔT_ml = @. ΔT_ml - (h_ice + Δh_ice) * p_i.L_ice / (p_o.h * p_o.ρ * p_o.c)
+        Δh_ice = @. -h_ice
+    end
+
+    # solve for T_sfc
+    if (parent(h_ice .+ Δh_ice)[1] > 0) #  surface is ice-covered
+        # if ice covered, solve implicity (for now one Newton iteration: ΔT_s = - F(T_s) / dF(T_s)/dT_s )
+        F_conductive = @. p_i.k_ice / (h_ice + Δh_ice) * (p_i.T_base - T_sfc)
+        ΔT_sfc = @. (-F_atm + F_conductive) / (p_i.k_ice / (h_ice + Δh_ice) + ∂F_atmo∂T_sfc)
```

Comment:
This is a little hard to read. But I understand there may be some restrictions from ClimaCore that prevent writing it a different way (just not sure)

Reply:
I added the clarifications of the if statement as you suggest above. Beyond that, I think it's ok, given that this will be a temporary toy model anyway. That said, would it be clearer to make pointers for the `p_i` and `p_o` parameters?

Reply:
No code is temporary! :-D

Reply:
I think dividing into a few lines with a bit of white space would be the main improvement, something like

```julia
numerator = -F_atm + F_conductive

h = h_ice + Δh_ice
k = p_i.k_ice
denominator = k / h + ∂F_atmo∂T_sfc

ΔT_sfc = @. numerator / denominator
```

Reply:
Haha, true true. And sounds good! Just revamped it!